packages/core/src/sync-realtime/service.ts

```diff
@@ -710,7 +716,74 @@ const getMatchedLogs = async (
       logs,
       logFilters: [
         ...service.logFilterSources.map((l) => l.criteria),
-        ...factoryLogFilters,
+        ...factoryLogFilters.filter((f) => f.address.length !== 0),
```

Comment:
I think I actually caught a subtle bug in the `getMatchedLogs` function here that would cause us to match on every "address" in the case when there were no child addresses in the database.